precog/validators/scripts/auto_updater.py

```
@@ -0,0 +1,100 @@
+import importlib.resources as pkg_resources
+import logging
+import subprocess
+import time
+from datetime import timedelta
+
+import git
+
+import precog
+from precog.utils.timestamp import elapsed_seconds, get_now
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+TIME_INTERVAL = 5
+
+
+def git_pull_change(path) -> bool:
+    repo = git.Repo(path)
+    current_hash = repo.head.commit
+
+    # Check if there are local changes
+    must_stash = repo.is_dirty()
+
+    # Stash the changes
+    if must_stash:
+        logger.info("Local changes detected. Stashing...")
+        repo.git.stash("save")
```

Won't using `stash`, then `pop` potentially ***mutate*** the Validator's local git status?

e.g. - if you had staged file modifications, then stashed, then popped, those modifications are now unstaged. folks might be wondering where their tracked changes went, if any.

should we instead just throw a useful error message if a Validator has uncommitted local code, ? `"You have local updates - please stash/commit/discard/.."`... this way, we put the responsibility on them to properly organize their work instead of us potentially borking it with code.

---

generally speaking, validators should not be altering files tracked by git. this stashing functionality will most likely go unused. i've added a disclaimer to our README indicating that validators who intend on changing the source code should be opting out of the auto update

---

removed stashing functionality for now. simply killing the process if the repo is dirty